src/components/AnonymousChat/child/ChatItemMyTextV2.tsx

```diff
@@ -59,31 +81,37 @@ const ChatItemMyTextV2 = ({
         <IconChatCheckMark color={colors.silver} width={12} height={12} />
       </View>
     );
-  }, []);
+  }, [status]);
 
-  const contextMenuActions = [
-    {title: 'Reply', systemIcon: 'arrow.turn.up.left'},
-    {title: 'Copy Message', systemIcon: 'square.on.square'},
-    {title: 'Delete Message', systemIcon: 'trash', destructive: true}
+  const contextMenuActions: ContextMenuAction[] = [
+    {title: CONTEXT_MENU_REPLY, systemIcon: 'arrow.turn.up.left'}
   ];
 
+  if (messageType !== MESSAGE_TYPE_DELETED) {
+    contextMenuActions.push({title: CONTEXT_MENU_COPY, systemIcon: 'square.on.square'});
+    contextMenuActions.push({title: CONTEXT_MENU_DELETE, systemIcon: 'trash', destructive: true});
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The context menu actions are now dynamically added based on the `messageType`. This is a good improvement for flexibility and maintainability. However, please make sure that the `CONTEXT_MENU_*` constants match the actual system icons' names. 



<!-- This is an auto-generated comment by OSS CodeRabbit -->